### PR TITLE
PN-351: Add comments field so clinicians/group managers can make a note as why the specific match was rejected or saved

### PR DESCRIPTION
**Pull request by veronikaslc**
Done.

### CODE REVIEW

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/MatchingNotificationManager.java [hunk 1]
```diff
@@ -111,4 +111,21 @@ boolean saveOutgoingMatches(List<? extends PatientSimilarityView> similarityView
...
      */
     boolean setStatus(Set<Long> matchesIds, String status);
+
+    /**
+     * Marks all matches with ids in {@code matchesIds} as notified.
+     *
+     * @param matchesIds list of ids of matches to mark as notified.
+     * @return true if successful
+     */
+    boolean markNotified(Set<Long> matchesIds);
```
[the remark this answers is not in the document]
Comment (veronikaslc) [reply matching-notification-api/src/main/java/org/phenotips/matchingnotification/MatchingNotificationManager.java]: just a tiny refactor of the implementation of previous method. Here I moved markNotified and made saveComment REST calls to both go through MatchingNotificationManager.

[the remark this answers is not in the document]
Comment (veronikaslc) [reply matching-notification-api/src/main/java/org/phenotips/matchingnotification/MatchingNotificationManager.java]: I'll extract this refactor into separate commit.

**Comment by sashaandjic** [comment]
Ideally we should have an indicator if there is a comment for each match. Can we show a different icon if there is a comment? Also, both the comment icon and the CONTACT and MARK CONTACTED buttons appear too big and very dominating on the screen. Can we go down with the size to improve the user experience?

File: matching-notification-api/src/main/java/org/phenotips/matchingnotification/MatchingNotificationManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (veronikaslc) [reply matching-notification-api/src/main/java/org/phenotips/matchingnotification/MatchingNotificationManager.java]: removed separated commit, changes are made in [PN-349 PR](https://github.com/phenotips/patient-network/pull/214)

**Comment by veronikaslc** [comment]
Added an indicator if there is a comment entered in last commit.

**Comment by veronikaslc** [comment]
Added.

**Comment by allasm** [comment]
Code reviewed/confirmed. As discussed, some CSS changes can be implemented to make it look better on smaller monitors

**Comment by veronikaslc** [comment]
addressed meeting's codereview comments and rebased on latest master

**Pull request merged**